File: src/cn/timelives/java/math/CalculatorHolder.java
/**
 * 2018-03-05
 */
package cn.timelives.java.math;

import cn.timelives.java.math.algebra.abstractAlgebra.calculator.EqualPredicate;

/**
 * @author liyicheng
 * 2018-03-05 20:25
 *
 */
public interface CalculatorHolder<T, S extends EqualPredicate<T>> {

	/**
	 * Return the calculator this object is using.
	 * 
	 * @return a calculator
	 */
	S getMathCalculator();
}

File: src/cn/ancono/math/CalculatorHolder.kt
/**
 * 2018-03-05
 */
package cn.ancono.math

import cn.ancono.math.algebra.abs.calculator.EqualPredicate

/**
 * @author liyicheng
 * 2018-03-05 20:25
 */
interface CalculatorHolder<T : Any, S : EqualPredicate<T>> {

    /**
     * Return the calculator used by this object.
     *
     * @return a calculator
     */
    val mathCalculator: S

}
